internal/cache/controller_obj_cache.go

```diff
@@ -2000,14 +2055,13 @@ func (c *AviObjCache) AviObjVSCachePopulate(client *clients.AviClient, cloud str
 							httpUuid := ExtractUuid(httpmap["http_policy_set_ref"].(string), "httppolicyset-.*.#")
 							httpName, foundhttp := c.HTTPPolicyCache.AviCacheGetNameByUuid(httpUuid)
 							// If the httppol is not found in the cache, do an explicit get
-							if !foundhttp && !sharedVsOrL4 && httpCacheRefreshCount > 0 {
-								// We do a full refresh of the httpcache once per page, if we detect a data discrepancy
-								httpCacheRefreshCount = httpCacheRefreshCount - 1
-								c.PopulateHttpPolicySetToCache(client, cloud)
-								httpName, foundhttp = c.HTTPPolicyCache.AviCacheGetNameByUuid(httpUuid)
-								if !foundhttp {
-									// If still the httpName is not found. Log an error saying, this VS may not behave appropriately.
+							if !foundhttp && !sharedVsOrL4 {
+								err = c.AviPopulateHttpPolicySetbyUUID(client, httpUuid)
+								// If still the httpName is not found. Log an error saying, this VS may not behave appropriately.
+								if err != nil {
```

Comment:
Should we not recalculate values of `httpName` and `foundhttp` after cache population here ?

Reply:
values won't change in error condition. In else condition it is handled.

Reply:
but after error we are populating httppolicyset by calling this function , `AviPopulateHttpPolicySetbyUUID`, will it not change values for httpName, foundhttp